docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+
+### SwiftNIO helper functions
+
+To allow an easy transition to async code, SwiftNIO offers a number of helper methods on `EventLoopFuture` and `-Promise`. Those live in the `_NIOConcurrency` module and will move to `NIOCore` once Swift concurrency is released.
+
+On every `EventLoopFuture` you can call `.get()` to transition the future into an `await`-able invocation. If you want to translate async/await calls to an `EventLoopFuture` we recommend the following pattern:
+
+```swift 
+#if swift(>=5.5)
+func yourAsyncFunctionConvertedToAFuture(on eventLoop: EventLoop) 
+    -> EventLoopFuture<Result> {
+    let promise = context.eventLoop.makePromise(of: Out.self)
+    promise.completeWithTask {
+        try await yourMethod(yourInputs)
+    }
+    return promise.futureResult
+}
+#endif
+```
+
+Further helpers exist for `EventLoopGroup`, `Channel`, `ChannelOutboundInvoker` and `ChannelPipeline`.
+
+### Sendable Checking
+
+> [SE-0302][SE-0302] introduced the `Sendable` protocol, which is used to indicate which types have values that can safely be copied across actors or, more generally, into any context where a copy of the value might be used concurrently with the original. Applied uniformly to all Swift code, `Sendable` checking eliminates a large class of data races caused by shared mutable state.
+>
+> -- from [Staging in Sendable checking][sendable-staging], which outlines the `Sendable` adoption plan for Swift 6.
+
+In the future we will see fully checked Swift concurrency. The language features to support this are the `Sendable` protocol and the `@Sendable` keyword for closures. Since sendable checking will break existing Swift code, a new major Swift version is required.
+
+To ease the transition to fully checked Swift code, it is possible to annotate your APIs with the `Sendable` protocol today.
+
+You can start adopting Sendable and getting appropriate warnings in Swift 5.5 already by passing the `-warn-concurrency` flag, you can do so in SwiftPM for the entire project like so:
+
+```
+swift build -Xswiftc -Xfrontend -Xswiftc -warn-concurrency
+```
+
+
+#### Sendable checking today
+
+Sendable checking is currently disabled in Swift 5.5(.0) because it was causing a number of tricky situations for which we lacked the tools to resolve.
+
+Most of these issues have been resolved on today’s `main` branch of the compiler, and are expected to land in the next Swift 5.5 releases. It may be worthwhile waiting for adoption until the next version(s) after 5.5.0.
+
+For example, one of such capabilities is the ability for tuples of `Sendable` types to conform to `Sendable` as well. We recommend holding off adoption of `Sendable` until this patch lands in Swift 5.5 (which should be relatively soon). With this change, the difference between Swift 5.5 with `-warn-concurrency` enabled and Swift 6 mode should be very small, and manageable on a case by case basis.
```

Comment:
It would be better to put an expected date in rather than relatively soon or this guide will not read well in a few months time.